File: feature/src/main/java/com/ortin/gesturetranslator/feature/repository/WordCompilerRepositoryImpl.java
package com.ortin.gesturetranslator.feature.repository;

import com.ortin.gesturetranslator.domain.repository.WordCompilerRepository;
import com.ortin.gesturetranslator.feature.managers.word_compiler.WordCompilerManager;

public class WordCompilerRepositoryImpl implements WordCompilerRepository {
    private final WordCompilerManager wordCompilerManager;

    public WordCompilerRepositoryImpl(WordCompilerManager wordCompilerManager) {
        this.wordCompilerManager = wordCompilerManager;
    }

    @Override
    public void addLetter(String letter) {
        wordCompilerManager.addLetter(letter);
    }

    @Override
    public void clearState() {
        wordCompilerManager.clearState();
    }

    @Override
    public String getWord() {
        return wordCompilerManager.getWord();
    }
}

File: feature/src/main/kotlin/com/ortin/gesturetranslator/feature/repository/WordCompilerRepositoryImpl.kt
package com.ortin.gesturetranslator.feature.repository

import com.ortin.gesturetranslator.domain.repository.WordCompilerRepository
import com.ortin.gesturetranslator.feature.managers.word_compiler.WordCompilerManager
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class WordCompilerRepositoryImpl @Inject constructor(private val wordCompilerManager: WordCompilerManager) :
    WordCompilerRepository {
    override fun addLetter(letter: String) {
        wordCompilerManager.addLetter(letter)
    }

    override fun clearState() {
        wordCompilerManager.clearState()
    }

    override fun getWord(): String {
        return wordCompilerManager.getWord()
    }
}
